Client/mods/deathmatch/logic/luadefs/CLuaDrawingDefs.cpp

```diff
@@ -1728,6 +1728,43 @@ int CLuaDrawingDefs::DxGetStatus(lua_State* luaVM)
         lua_pushnumber(luaVM, static_cast<lua_Number>(SharedUtil::GetWMITotalPhysicalMemory()) / 1024.0 / 1024.0);
         lua_settable(luaVM, -3);
 
+        // diagnostic debug detection
+        
+        EDiagnosticDebugType diagnosticDebug = g_pCore->GetDiagnosticDebug();
+        std::string diagnosticDebugString;
+
+        switch (diagnosticDebug)
+        {
+            case EDiagnosticDebug::GRAPHICS_6734:
+                diagnosticDebugString = "#6734 Graphics";
+                break;
+            case EDiagnosticDebug::D3D_6732:
+                diagnosticDebugString = "#6732 D3D";
+                break;
+            case EDiagnosticDebug::LOG_TIMING_0000:
+                diagnosticDebugString = "#0000 Log timing";
+                break;
+            case EDiagnosticDebug::JOYSTICK_0000:
+                diagnosticDebugString = "#0000 Joystick";
+                break;
+            case EDiagnosticDebug::LUA_TRACE_0000:
+                diagnosticDebugString = "#0000 Lua trace";
+                break;
+            case EDiagnosticDebug::RESIZE_ALWAYS_0000:
+                diagnosticDebugString = "#0000 Resize always";
+                break;
+            case EDiagnosticDebug::RESIZE_NEVER_0000:
+                diagnosticDebugString = "#0000 Resize never";
+                break;
+            default:
+                diagnosticDebugString = "Default";
+                break;
+        }
+
+        lua_pushstring(luaVM, "SettingDebugMode");
+        lua_pushstring(luaVM, diagnosticDebugString.c_str());
```

Comment:
```suggestion
              lua::Push(luaVM, "SettingDebugMode");
        lua::Push(luaVM, []{
            switch (g_pCore->GetDiagnosticDebug())
            {
            case EDiagnosticDebug::GRAPHICS_6734:
                return "#6734 Graphics";
            case EDiagnosticDebug::D3D_6732:
                return "#6732 D3D";
            case EDiagnosticDebug::LOG_TIMING_0000:
                return "#0000 Log timing";
            case EDiagnosticDebug::JOYSTICK_0000:
                return "#0000 Joystick";
            case EDiagnosticDebug::LUA_TRACE_0000:
                return "#0000 Lua trace";
            case EDiagnosticDebug::RESIZE_ALWAYS_0000:
                return "#0000 Resize always";
            case EDiagnosticDebug::RESIZE_NEVER_0000:
                return "#0000 Resize never";
            default:
                return "Default";
            }
        }());
```

Also, side-note, instead of default maybe we should return `None` or something similar.